### PR TITLE
Add MDP controller to asmcli.

### PR DESCRIPTION
**Pull request by bianpengyuan**
cc @mandarjog @ostromart @therealmitchconnors

### CODE REVIEW

**Comment by zerobfd** [comment]
FYI, the test failure is yamllint

```
../asm/istio/options/cni-managed.yaml
  96:1      error    wrong indentation: expected 2 but found 0  (indentation)
  97:3      error    wrong indentation: expected 4 but found 2  (indentation)
  99:3      error    wrong indentation: expected 4 but found 2  (indentation)
  101:3     error    wrong indentation: expected 4 but found 2  (indentation)
  103:3     error    wrong indentation: expected 4 but found 2  (indentation)
  106:3     error    wrong indentation: expected 4 but found 2  (indentation)
  113:3     error    wrong indentation: expected 4 but found 2  (indentation)
  116:3     error    wrong indentation: expected 4 but found 2  (indentation)
  118:3     error    wrong indentation: expected 4 but found 2  (indentation)
  121:3     error    wrong indentation: expected 4 but found 2  (indentation)
  123:3     error    wrong indentation: expected 4 but found 2  (indentation)
  125:3     error    wrong indentation: expected 4 but found 2  (indentation)
  127:3     error    wrong indentation: expected 4 but found 2  (indentation)
  129:3     error    wrong indentation: expected 4 but found 2  (indentation)
  131:3     error    wrong indentation: expected 4 but found 2  (indentation)
  134:3     error    wrong indentation: expected 4 but found 2  (indentation)
  146:1     error    wrong indentation: expected 2 but found 0  (indentation)
  163:81    warning  line too long (87 > 80 characters)  (line-length)
  221:81    warning  line too long (94 > 80 characters)  (line-length)
  222:81    warning  line too long (95 > 80 characters)  (line-length)
  228:81    warning  line too long (117 > 80 characters)  (line-length)
  257:81    warning  line too long (117 > 80 characters)  (line-length)
  280:81    warning  line too long (109 > 80 characters)  (line-length)
  316:11    error    wrong indentation: expected 12 but found 10  (indentation)
  329:13    error    wrong indentation: expected 14 but found 12  (indentation)
```

**Comment by zerobfd** [comment]
LGTM, I'll approve after one of @mandarjog @ostromart @therealmitchconnors approve.

File: asm/istio/options/cni-managed.yaml
Comment (mandarjog) [inline asm/istio/options/cni-managed.yaml:295]: why do we need to run as 1337?

Comment (mandarjog) [reply asm/istio/options/cni-managed.yaml:295]: is it just because of the base image?

Comment (bianpengyuan) [reply asm/istio/options/cni-managed.yaml:295]: All istio container, including istiod, runs as 1337. I guess it is a convention..

**Comment by ostromart** [comment]
Can we also add the DataPlaneControl CRD and dummy CR in here?

**Comment by bianpengyuan** [comment]
> Can we also add the DataPlaneControl CRD and dummy CR in here?

Ok will do once it is ready.

**Comment by bianpengyuan** [comment]
@zerobfd can we disable line length check of yamllint, or at least increase the limit? Got whole bunch of line too long error after adding CRD:
```
  44:81     error    line too long (83 > 80 characters)  (line-length)
  47:81     error    line too long (86 > 80 characters)  (line-length)
  48:81     error    line too long (81 > 80 characters)  (line-length)
  49:81     error    line too long (169 > 80 characters)  (line-length)
  52:81     error    line too long (82 > 80 characters)  (line-length)
  53:81     error    line too long (82 > 80 characters)  (line-length)
  54:81     error    line too long (173 > 80 characters)  (line-length)
  59:81     error    line too long (83 > 80 characters)  (line-length)
  63:81     error    line too long (82 > 80 characters)  (line-length)
  64:81     error    line too long (83 > 80 characters)  (line-length)
  65:81     error    line too long (82 > 80 characters)  (line-length)
  66:81     error    line too long (83 > 80 characters)  (line-length)
  67:81     error    line too long (90 > 80 characters)  (line-length)
  68:81     error    line too long (85 > 80 characters)  (line-length)
  69:81     error    line too long (82 > 80 characters)  (line-length)
  74:81     error    line too long (81 > 80 characters)  (line-length)
  78:81     error    line too long (84 > 80 characters)  (line-length)
  83:81     error    line too long (86 > 80 characters)  (line-length)
  102:81    error    line too long (87 > 80 characters)  (line-length)
```

**Comment by bianpengyuan** [comment]
Here is the yamllint configuraiton we use in istio: https://github.com/istio/istio/blob/master/common/config/.yamllint.yml where line length is disabled.

**Comment by bianpengyuan** [comment]
@ostromart please take another look. I added CRD. Other than lint error I think this is ready for review.

File: asm/istio/options/cni-managed.yaml
Comment (bianpengyuan) [inline asm/istio/options/cni-managed.yaml:15]: @kyessenov I am getting warning when applying:
```
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
```

I saw some discussion about this and I assume this is expected?

Comment (mandarjog) [reply asm/istio/options/cni-managed.yaml:15]: if all other crds use v1beta1 then we will do en-mass migration. Since this is syncd internally changing it here will require another sync.

File: asm/istio/options/cni-managed.yaml
Comment (mandarjog) [inline asm/istio/options/cni-managed.yaml:21]: Is this the correct order?
Revision should be first.

File: asm/istio/options/cni-managed.yaml
Comment (mandarjog) [reply asm/istio/options/cni-managed.yaml:399]: The base image has this uid, ok.

**Review by mandarjog — DISMISSED** [review]
Lgtm sans column ordering.
But that can follow later.

File: asm/istio/options/cni-managed.yaml
Comment (bianpengyuan) [reply asm/istio/options/cni-managed.yaml:21]: I copied it from https://gke-internal-review.googlesource.com/c/istio/istio/+/297502. @kyessenov ^^

File: asm/istio/options/cni-managed.yaml
Comment (kyessenov) [reply asm/istio/options/cni-managed.yaml:15]: This is tracked internally. You can disregard the warning since it's g3 wide.

File: asm/istio/options/cni-managed.yaml
Comment (kyessenov) [reply asm/istio/options/cni-managed.yaml:21]: I wasn't aware revision was first. I requested to add revision in CL.

Comment (richardwxn) [reply asm/istio/options/cni-managed.yaml:21]: I created cl in g3 to reorder it as: revision, version, status and age

Comment (bianpengyuan) [reply asm/istio/options/cni-managed.yaml]: Will update once the internal CL lands.

**Comment by zerobfd** [comment]
> Here is the yamllint configuraiton we use in istio: https://github.com/istio/istio/blob/master/common/config/.yamllint.yml where line length is disabled.

@bianpengyuan Feel free to change the `asmcli/.yamllint` to allow 120 char lines

**Review by zerobfd — APPROVED** [review]

**Pull request merged**
